lib/minify.js

```diff
@@ -270,6 +278,24 @@ async function minify(files, options, _fs_module) {
         result.ast = toplevel.to_mozilla_ast();
     }
     if (!HOP(options.format, "code") || options.format.code) {
+        if (!options.format.ast) {
+            // Destroy stuff to save RAM. (unless the deprecated `ast` option is on)
+            options.format._destroy_ast = true;
+
+            walk(toplevel, node => {
+                if (node instanceof AST_Scope) {
+                    node.variables = undefined;
+                    node.enclosed = undefined;
+                    node.parent_scope = undefined;
+                }
+                if (node.block_scope) {
+                    node.block_scope.variables = undefined;
+                    node.block_scope.enclosed = undefined;
+                    node.parent_scope = undefined;
+                }
+            });
```

Comment:
This is done before the `options.sourceMap.includeSources` check below, which increases memory usage.